src/Microsoft.Health.Dicom.Api/Web/SeekableStreamConverter.cs

```diff
@@ -21,37 +22,54 @@ internal class SeekableStreamConverter : ISeekableStreamConverter
 {
     private const int DefaultBufferThreshold = 1024 * 30000; // 30MB
     private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly ILogger<SeekableStreamConverter> _logger;
 
-    public SeekableStreamConverter(IHttpContextAccessor httpContextAccessor)
+    public SeekableStreamConverter(IHttpContextAccessor httpContextAccessor, ILogger<SeekableStreamConverter> logger)
     {
-        EnsureArg.IsNotNull(httpContextAccessor, nameof(httpContextAccessor));
-
-        _httpContextAccessor = httpContextAccessor;
+        _httpContextAccessor = EnsureArg.IsNotNull(httpContextAccessor, nameof(httpContextAccessor));
+        _logger = EnsureArg.IsNotNull(logger, nameof(logger));
     }
 
     /// <inheritdoc />
     public async Task<Stream> ConvertAsync(Stream stream, CancellationToken cancellationToken = default)
     {
-        EnsureArg.IsNotNull(stream, nameof(stream));
+        try
+        {
 
-        int bufferThreshold = DefaultBufferThreshold;
-        long? bufferLimit = null;
-        Stream seekableStream = null;
+            EnsureArg.IsNotNull(stream, nameof(stream));
+
+            int bufferThreshold = DefaultBufferThreshold;
+            long? bufferLimit = null;
+            Stream seekableStream = null;
+
+            if (!stream.CanSeek)
+            {
+                seekableStream = new FileBufferingReadStream(stream, bufferThreshold, bufferLimit, AspNetCoreTempDirectory.TempDirectoryFactory);
+                _httpContextAccessor.HttpContext?.Response.RegisterForDisposeAsync(seekableStream);
+                await seekableStream.DrainAsync(cancellationToken);
+            }
+            else
+            {
+                seekableStream = stream;
+            }
+
+            seekableStream.Seek(0, SeekOrigin.Begin);
+
+            return seekableStream;
 
-        if (!stream.CanSeek)
-        {
-            seekableStream = new FileBufferingReadStream(stream, bufferThreshold, bufferLimit, AspNetCoreTempDirectory.TempDirectoryFactory);
-            _httpContextAccessor.HttpContext?.Response.RegisterForDisposeAsync(seekableStream);
-            await seekableStream.DrainAsync(cancellationToken);
         }
-        else
+        catch (InvalidMultipartBodyPartException ex)
         {
-            seekableStream = stream;
-        }
+            _logger.LogWarning(ex, "Unexpected end of the stream. This is likely due to the request being multipart but has no section.");
 
-        seekableStream.Seek(0, SeekOrigin.Begin);
+            throw;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogWarning(ex, "Unhandled exception while reading stream.");
```

Comment:
I don't think we need to catch the general exception. If we only catch `InvalidMultipartBodyPartException`, then perhaps we can narrow the scope of the try block to the relevant bits